From 555c1919bf06032f6840a6e061f636ab4213cb58 Mon Sep 17 00:00:00 2001
From: Saahil Tomar <quic_saahtoma@quicinc.com>
Date: Tue, 23 Jan 2024 12:15:19 +0530
Subject: [PATCH] bus: mhi: Fix KW issue

This change initializes the value of rx_status in
mhi_download_rddm_image function.

Change-Id: I2bcdc57b8156467ee47722b5be9c7e5ce588cff2
Signed-off-by: Saahil Tomar <quic_saahtoma@quicinc.com>
(cherry picked from commit aafa03b88a1c790139db5abe90bd836dad346810)
---
 drivers/bus/mhi/host/boot.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/bus/mhi/host/boot.c b/drivers/bus/mhi/host/boot.c
index d95bf9ede967..9e664799b99a 100644
--- a/drivers/bus/mhi/host/boot.c
+++ b/drivers/bus/mhi/host/boot.c
@@ -189,7 +189,7 @@ int mhi_download_rddm_image(struct mhi_controller *mhi_cntrl, bool in_panic)
 	struct device *dev = &mhi_cntrl->mhi_dev->dev;
 	rwlock_t *pm_lock = &mhi_cntrl->pm_lock;
 	struct mhi_buf *mhi_buf = NULL;
-	u32 rx_status;
+	u32 rx_status = 0;
 	int ret;
 
 	/*
-- 
2.34.1

